### PR TITLE
Fix `latest_execution_payload_header` field in eip4844 state upgrade 

### PR DESCRIPTION
**Pull request by tbenr**
I think latest_execution_payload_header upgrade has been forgotten

### CODE REVIEW

File: specs/eip4844/fork.md [hunk 1]
```diff
@@ -66,8 +66,25 @@ Since the `eip4844.BeaconState` format is equal to the `bellatrix.BeaconState` f
...
+        prev_randao=pre.latest_execution_payload_header.prev_randao,
+        block_number=pre.latest_execution_payload_header.block_number,
+        gas_limit=pre.latest_execution_payload_header.gas_limit,
+        gas_used=pre.latest_execution_payload_header.gas_used,
+        timestamp=pre.latest_execution_payload_header.timestamp,
+        extra_data=pre.latest_execution_payload_header.extra_data,
+        base_fee_per_gas=pre.latest_execution_payload_header.base_fee_per_gas,
+        excess_blobs=uint64(0),     # [New in EIP-4844]
+        block_hash=pre.latest_execution_payload_header.block_hash,
+        transactions_root=pre.latest_execution_payload_header.transactions_root,
```
Comment (ralexstokes) [inline specs/eip4844/fork.md:86]:
```suggestion
        transactions_root=pre.latest_execution_payload_header.transactions_root,
        withdrawals_root=pre.latest_execution_payload_header.withdrawals_root,
```

File: specs/eip4844/fork.md [hunk 2]
```diff
@@ -66,8 +66,25 @@ Since the `eip4844.BeaconState` format is equal to the `bellatrix.BeaconState` f
...
+        receipts_root=pre.latest_execution_payload_header.receipts_root,
+        logs_bloom=pre.latest_execution_payload_header.logs_bloom,
+        prev_randao=pre.latest_execution_payload_header.prev_randao,
+        block_number=pre.latest_execution_payload_header.block_number,
+        gas_limit=pre.latest_execution_payload_header.gas_limit,
+        gas_used=pre.latest_execution_payload_header.gas_used,
+        timestamp=pre.latest_execution_payload_header.timestamp,
+        extra_data=pre.latest_execution_payload_header.extra_data,
+        base_fee_per_gas=pre.latest_execution_payload_header.base_fee_per_gas,
+        excess_blobs=uint64(0),     # [New in EIP-4844]
```
Comment (ralexstokes) [inline specs/eip4844/fork.md:84]:
```suggestion
        excess_data_gas=uint256(0),     # [New in EIP-4844]
```

**Review by ralexstokes — COMMENTED** [review]
nice catch! 

#3044 is related

**Comment by tbenr** [comment]
since #3052 is not yet merged, I was proposing a change to the current version, and let the rebase change the header here.

**Review by hwwhww — APPROVED** [review]
Thanks @tbenr!

I merged @ralexstokes's suggestions and think we can go merge this PR first.

note: also related to #3004

**Pull request merged**